src/main/java/Duke.java

```diff
@@ -1,10 +1,146 @@
+        System.out.println("_____________________________");
+
+        // to read input on each new line, Duke constantly scans input in this loop
+        Scanner sc = new Scanner(System.in);
+        int taskNumber = 0;
+        TaskManager manager = new TaskManager(NUMBER_OF_TASKS);
+        while(true) {
+            String inputStr = sc.nextLine();
+
+            if (getCommand(inputStr).equals("bye")) {
+                // exit
+                System.out.println("Bye. Hope to see you again soon!");
+                System.out.println("_____________________________");
+                break;
+            } else if (getCommand(inputStr).equals("list")) {
+                TaskManager.printTaskList();
+            } else if (getCommand(inputStr).contains("done")) {
+                if (isEmptyItem(inputStr)) {
+                    System.out.println("Oops, invalid description!");
+                    System.out.println("_____________________________");
+                    continue;
+                } else if (TaskManager.getNumberOfTasksUndone() == 0
+                        | TaskManager.getNumberOfTasksUndone() < Integer.parseInt(getItem(inputStr))) {
+                    System.out.println("Oops, invalid description");
+                    System.out.println("_____________________________");
+                    continue;
+                }
+                TaskManager.markTaskAsDone(inputStr);
+            } else if (getCommand(inputStr).contains("todo")) {
+                if (isEmptyItem(inputStr)) {
+                    System.out.println("Oops, invalid description!");
+                    System.out.println("_____________________________");
+                    continue;
+                }
+                String item = getItem(inputStr);
+                TaskManager.addToDoTaskToList(item, taskNumber);
+                taskNumber++;
+            } else if (getCommand(inputStr).contains("deadline")) {
+                if (isInvalidItem(inputStr)) {
+                    System.out.println("Oops, invalid description!");
+                    System.out.println("_____________________________");
+                    continue;
+                }
+                String item = getItem(inputStr);
+                TaskManager.addDeadlineTaskToList(item, taskNumber);
+                taskNumber++;
+            } else if (getCommand(inputStr).contains("event")) {
+                if (isInvalidItem(inputStr)) {
+                    System.out.println("Oops, invalid description!");
+                    System.out.println("_____________________________");
+                    continue;
+                }
+                String item = getItem(inputStr);
+                TaskManager.addEventTaskToList(item, taskNumber);
+                taskNumber++;
+            } else {
+                System.out.println("Oops, command not recognised!");
+                System.out.println("_____________________________");
```

Comment:
Can declare both strings as constants to make code look neater, since it is commonly used

Reply:
thank you for your input. will change accordingly!